components/search-filters/filter.tsx

```diff
@@ -1,74 +1,83 @@
 import {
   buildSearchQuery,
   IParams,
   ISearchFilter,
 } from '#models/search-filter-params';
-import { randomId } from '#utils/helpers';
+import { useOutsideClick } from 'frontend/src/hooks';
 import ActiveFilterLabel from './active-filter-label';
 
-const Filter: React.FC<
-  PropsWithChildren<{
-    label: string;
-    activeFilter: ISearchFilter;
-    searchParams: IParams;
-    searchTerm: string;
-    addSaveClearButton: boolean;
-  }>
-> = ({
+type FilterProps = {
+  children: ReactElement[];
+  label: string;
+  activeFilter: ISearchFilter;
+  searchParams: IParams;
+  searchTerm: string;
+  addSaveClearButton: boolean;
+};
+
+const Filter = ({
   children,
   label,
   activeFilter,
   searchParams,
   searchTerm,
   addSaveClearButton = false,
-}) => {
-  const uuid = `toggle-${randomId()}`;
-
+}: FilterProps) => {
   const clearFilterLink = buildSearchQuery(
     searchTerm,
     searchParams,
     activeFilter.excludeParams
   );
 
+  const [open, setOpen] = useState(false);
+
+  const ref = useOutsideClick(() => {
+    setOpen(false);
+  });
+
   return (
     <>
       <div className="search-filter-label-container">
-        <input type="checkbox" id={uuid} />
-        <label htmlFor={uuid} className="overlay" />
-        <label htmlFor={uuid}>
-          {activeFilter.label ? (
-            <ActiveFilterLabel
-              icon={activeFilter.icon}
-              label={activeFilter.label}
-              query={clearFilterLink}
-            />
-          ) : (
-            <span className="search-filter-label">
-              {activeFilter.icon}&nbsp;{label}&nbsp;&nbsp;▾
-            </span>
-          )}
-        </label>
-        <label htmlFor={uuid} className="close-container">
-          Fermer ✕
-        </label>
-        <div className="container">
-          <div className="filter-container">{children}</div>
-          {addSaveClearButton && (
-            <>
-              <br />
-              <div className="layout-space-between">
-                <a className="fr-link" href={clearFilterLink}>
-                  Effacer
-                </a>
-                <ButtonLink type="submit" alt small>
-                  Appliquer
-                </ButtonLink>
-              </div>
-            </>
+        <div>
+          <label onClick={() => setOpen(!open)}>
+            {activeFilter.label ? (
+              <ActiveFilterLabel
+                icon={activeFilter.icon}
+                label={activeFilter.label}
+                query={clearFilterLink}
+              />
+            ) : (
+              <span className="search-filter-label">
+                {activeFilter.icon}&nbsp;{label}&nbsp;&nbsp;▾
+              </span>
+            )}
+          </label>
+          {open && (
+            <label onClick={() => setOpen(false)} className="close-container">
+              Fermer ✕
+            </label>
           )}
         </div>
+        {open && (
+          <div className="container" ref={ref}>
```

Comment:
`<div className="container" ref={ref} style={{ display:open ? 'block' : 'none'}}>`

Should fix issue related to submit